nntrainer/layers/layer_context.cpp

```diff
@@ -14,10 +14,20 @@
 #include <functional>
 
 #include <layer_context.h>
+#include <stdexcept>
 #include <var_grad.h>
 #include <weight.h>
 
 namespace nntrainer {
+void InitLayerContext::setOutputDimensions(
```

Comment:
out of scope for this PR but layer context must add layer name to its error messages for debugging purposes.

Reply:
Let's do this in other PR by unifying NNTR_THROW_IF contains __func__ and __LINE__  + each source file (optionally) have additional and common information debug info then updating every place by hand.

Reply:
The layer name information + how to fix this will have to added by hand :)

Reply:
we can't tell how to fix this at this function, but added name here :)